projects/demo/src/app/demo/pages/scroll-pages/scroll-page.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { data } from './data_1189';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+export interface Data {
+    _id: string;
+    index: number;
+    guid: string;
+    name: string;
+    company: string;
+    email: string;
+    phone: string;
+    address: string;
+    about: string;
+}
+
+export interface Pagination {
+    total: number;
+    hasMore: boolean;
+}
+
+export interface DataPage {
+    data: Data[];
+    pagination: Pagination;
+}
+
+@Injectable({ providedIn: 'root' })
+export class ScrollPageService {
+    private readonly _batchSize = 25;
+
+    private _loadedData: BehaviorSubject<DataPage> = new BehaviorSubject<DataPage>({
+        data: data.slice(0, this._batchSize),
+        pagination: { total: data.length, hasMore: true },
+    });
+
+    private _loadingMore = false;
+
+    getData(): Observable<DataPage> {
+        return this._loadedData.asObservable();
+    }
+
+    loadMore(lastIndex: number) {
+        if (!this._loadingMore) {
+            console.log(`Load more from ${lastIndex}`);
```

Comment:
Do we want this console.log to stay because it's demo purpose?

Reply:
yes, I did let it for the demo, thinking it might help understanding what was going on in the demo while scrolling. Do you think I should remove it?

Reply:
No, it's fine.